Foundation/NSPredicate.swift

```diff
@@ -196,9 +196,9 @@ extension NSMutableSet {
 
 extension NSOrderedSet {
     open func filtered(using predicate: NSPredicate) -> NSOrderedSet {
-        return NSOrderedSet(array: self.allObjects.filter({ object in
+        return NSOrderedSet(array: self.array.filter { object in
```

Comment:
why go through the snapshot (and thus bridged) array here?

Reply:
I took this portion from the other PR - I'll take another look for better options

Reply:
I can't see a way around group through bridging, since the initializers to NSOrderedSet accept `[Any]`

I may be misunderstanding this though. Do you have any suggestions on a better way to do this?

Reply:
I was thinking just using the ivar -- but I see that this in an extension and perhaps doesn't have access.   The self.array will force a bridge from the NS type to the swift type, and so there's an extra semantic copy here (its perhaps not as bad as on Darwin platforms where we're crossing the objc->swift divide).

Reply:
I added back the `allObjects` accessor, so this change is no longer present